### DISCUSSION TITLE
Unexpected auto-installation of extensions in fresh Codespace

### DISCUSSION THREAD

**Discussion by oliversalzburg** [community/community#42225] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

I just logged into this GitHub account on this machine for the first time, and then I created a codespace in <https://github.com/kitten-science/kitten-scientists>. During the creation process, a bunch of VS Code extensions were installed, even though they are nowhere configured to be installed. When I went through the same process with a different account earlier, no extensions were installed.

As the extensions seemed familiar from past work, my first assumption was that this was an undesired side-effect of settings sync, but settings sync is disabled and can only be enabled:
![image](https://user-images.githubusercontent.com/1658949/208729140-358b893a-7cf6-462d-86ae-aef1c755afd0.png)

I definitely did not confirm any prompts to install anything and the behavior is entirely unexpected. What is going on?

**Comment by cmbrose** [comment]
👋 Thanks for reaching out @oliversalzburg! This isn't expected behavior and so we're trying to debug to see why you're seeing it. Would you mind filing a [support ticket](https://support.github.com/contact/bug-report) so that we can get more details from you, outside the public discussion?

If you could include the following in the ticket that would help us dig in immediately:
- Does this happen in web or desktop (or both)?
- If desktop, what is your vscode and codespaces extension versions?
- What are the extensions which were installed?
- Can you also run `devcontainer-info` in a terminal in the codespace and provide the output?

Thanks!

**Comment by spaltrowitz** [comment, marked as answer]
Hi @oliversalzburg, thanks for reporting this bug. A fix has now been shipped! 🚢 https://github.blog/changelog/2023-05-04-codespaces-settings-sync-updates/ cc @cmbrose

**Reply by oliversalzburg** [reply]
Oh nice. I'll familiarize myself with the new behavior :)

**Reply by N-Schaef** [reply]
I still have this problem.
Multiple times now I have turned off settings sync (and ticked the checkmark to delete the cloud data).
In the settings sync dialog I can see that my list of extensions is nearly empty (except for the default vscode ones).

I have around 5 local VSCode extensions installed on my local machine.

But everytime I create a new codespace around 30 old extensions that I desperately try to get rid of are getting installed.
Is this a result of having other old codespaces running that I did not yet clean?
